tm2py/components/network/highway/highway_assign.py

```diff
@@ -126,7 +126,7 @@ def validate_inputs(self):
     def run(self):
         """Run highway assignment."""
         demand = PrepareHighwayDemand(self.controller)
-        if self.controller.iteration >= 0:
+        if self.controller.iteration >= 0 or self.controller.config.run.warmstart:
```

Comment:
The second half of this if condition is not effective, as `self.controller.iteration >=0` is always true.

Instead of changing this line, I suggest you change the following section, add a warmstart condition to use non-zero demand:
https://github.com/camsys/tm2py/blob/304c7df647d751fe162749972f6e5ea1cf7bce73/tm2py/components/network/highway/highway_assign.py#L490-L496

Reply:
Addressed.